enterprise/server/remote_execution/containers/ociruntime/ociruntime.go

```diff
@@ -504,9 +516,19 @@ func (c *ociContainer) Exec(ctx context.Context, cmd *repb.Command, stdio *inter
 	}
 	args = append(args, c.cid)
 
-	return c.doWithStatsTracking(ctx, func(ctx context.Context) *interfaces.CommandResult {
+	res := c.doWithStatsTracking(ctx, func(ctx context.Context) *interfaces.CommandResult {
 		return c.invokeRuntime(ctx, cmd, stdio, 1*time.Microsecond, args...)
 	})
+
+	c.mu.Lock()
+	removed := c.removed
+	c.mu.Unlock()
+	if removed && (res.ExitCode == ociSIGKILLExitCode || res.ExitCode == ociExecSIGTERMExitCode) {
```

Comment:
in practice, I think `Exec()` and `Remove()` can't be called concurrently - `TracedCommandContainer` (which wraps all container instances) takes an exclusive lock before it calls `Remove()`